config/auth.js

```diff
@@ -107,13 +30,30 @@ module.exports = function (passport) {
 			(req, accessToken, refreshToken, profile, done) => {
 				// check if user belongs to UoL team
 				if (profile.team.id != process.env.SLACK_TEAM) {
-					return done(null, false, req.flash("profileMessage", "Please verify using UoL slack workspace"));
+					return done(null, false, req.flash("error", "Please verify using UoL slack workspace"));
 				}
-				db.query("UPDATE users SET slackuid= ?, verified=? WHERE id= ?", [profile.id, 1, req.user.id], (err, result) => {
+				db.query("SELECT * from users WHERE slackuid = ?", [profile.id], (err, result) => {
 					if (err) {
-						done(null, false, req.flash("profileMessage", "Could not verify profile or duplicate account"));
+						req.flash("error", "Slack login failed");
+						done(null, false);
+					} else if (result.length == 1) {
+						// known user. Return user id for deserializing
+						done(null, { id: result[0].id });
 					} else {
-						done(null, profile, req.flash("profileMessage", "Profile verified"));
+						// Insert new user
+						db.query(
+							"INSERT INTO users (username, email, slackuid) \
+                                values (?, ?, ?)",
+							[profile.displayName, profile.user.email, profile.id],
+							function (err, result) {
+								if (err) {
+									done(null, false);
+									req.flash("error", "Slack login failed");
+								}
+								// return user id for deserializing.
+								done(err, { id: result.insertId });
```

Comment:
My installation breaks here as it seems that "result" is undefined. The script to add a dummy user also fails with "Field 'password' doesn't have a default value"

Reply:
I have not run into any issues so far with this after rebuilding the db and changing env variables. Logged in and performed basic tasks successfully.

Reply:
Resolved after updating my env - didn't realise the DB is renamed.